### PR TITLE
Claim shopping list

### PR DESCRIPTION
**Pull request by Tetre2**

### CODE REVIEW

File: Project/routes/claimedLists.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    vego: 'on',
+    lowPrice: 'on',
+    list: { id: '1', name: 'ListLocal', items: [
+	   {product: "Palt",amount: 3}
+	   ]
+	   }
+  }
+]
+
+router.post('/:listId', function (req, res) {
```
Comment (Tetre2) [inline Project/routes/claimedLists.js:26]: remove /:listId since it is not used

File: Project/routes/publishedLists.js [hunk 2]
```diff
@@ -5,15 +5,45 @@ var router = express.Router();
...
 router.post('/:userId/publish', function (req, res) {
-	var item = req.body.publishedList;
-	publishedLists.push(item);
+	var list = req.body.publishedList;
+	list.id = findUniqueID(publishedLists);
+	publishedLists.push(list);
 	res.send('Received');
-  console.log("Added shoppinglist");
-  console.log(item);
+  console.log(`Published list "${list.id}" `);
```
Comment (Tetre2) [inline Project/routes/publishedLists.js:16]: is userId used? dosent seem like that. should it be there then?

File: Project/public/claimedLists.html [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+		<link rel="stylesheet" type="text/css" href="stylesheets/listElement.css">
+		<title>Helper-shopper</title>
+		<script type="text/javascript" src="javascripts/apiCaller.js"></script>
+		<script type="text/javascript" src="javascripts/claimedLists-frontend.js"></script>
+		<script type="text/javascript" src="javascripts/listElementGenerator.js"></script>
+	</head>
+	<body>
+		<header>
+			<section class="header-top">
+				<a href="index.html" class="back-button"> <svg width="100%" height="100%" viewBox="0 0 15 24" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (Eyzic) [inline Project/public/claimedLists.html:17]: lägg svg i en separat fil och ladda in som img

File: Project/public/createShoppingList.html [hunk 4]
```diff
@@ -12,10 +12,10 @@
     <main>
         <section class="nav-filter">
             <a href="index.html" class="nav-element"><button class="back-button"> &lt- </button></a>
-			<form action="javascript:updateCurrentListName();" class="nav-element main-searchBox">
+			         <form action="javascript:updateCurrentListName();" class="nav-element main-searchBox">
             <input id="listName" disabled >
 			</form>
-			<button class="edit-listname nav-element" onclick="var name = document.getElementById('listName'); 
+			<button class="edit-listname nav-element" onclick="var name = document.getElementById('listName');
```
Comment (Eyzic) [inline Project/public/createShoppingList.html:18]: behöver inte spara variabeln här. anropa disabled direkt.

File: Project/public/createShoppingList.html [hunk 5]
```diff
@@ -12,10 +12,10 @@
     <main>
         <section class="nav-filter">
             <a href="index.html" class="nav-element"><button class="back-button"> &lt- </button></a>
-			<form action="javascript:updateCurrentListName();" class="nav-element main-searchBox">
+			         <form action="javascript:updateCurrentListName();" class="nav-element main-searchBox">
```
Comment (Eyzic) [inline Project/public/createShoppingList.html:15]: konstig indentering?

File: Project/public/javascripts/availableOrders.js [hunk 6]
```diff
@@ -1,37 +1,39 @@
...
-      //Iterate through list and create a li element for each item.
-      for (x in publishedLists[y].items) {
-        var newItem = document.createElement("li");
-        var item = publishedLists[y].items[x];
-
-        newItem.innerHTML = `${item.product} - ${item.amount} st`;
-        items.appendChild(newItem);
+   requestOnLocalHost('GET', '/publishedLists/1', function (result) {
+
+		var resultArea = document.getElementById("js-displayed-lists");
```
Comment (Eyzic) [inline Project/public/javascripts/availableOrders.js:6]: konstig indentering

File: Project/routes/claimedLists.js (hunk 1 above)
Comment (Eyzic) [reply Project/routes/claimedLists.js:26]: or we could rename in anticipation for the method getAllClaimedLists.

File: Project/routes/publishedLists.js [hunk 7]
```diff
@@ -5,15 +5,45 @@ var router = express.Router();
 
 router.get('/:userId', function (req, res) {
 	res.send(JSON.stringify(publishedLists));
-	console.log(publishedLists);
+	console.log('Server sent the publishedlist');
```
Comment (Eyzic) [inline Project/routes/publishedLists.js:8]: Could have a nicer log that actually tells us what was sent.

File: Project/routes/publishedLists.js (hunk 2 above)
Comment (Eyzic) [reply Project/routes/publishedLists.js:16]: now that you mention it, we actually dont need it since the id is unique for published Lists.

**Pull request merged**